src/app/centroidSwitch.js

```diff
@@ -0,0 +1,163 @@
+
+          if (!this.enabled) {
+              return;
+          }
+
+          // if they are just panning don't worry about it
+          // if the delta is null, it is the first load so
+          // we need to show the centroids
+          if (!extent.levelChange && extent.delta) {
+              return;
+          }
+
+          this._loadLayers().then(lang.hitch(this, function () {
+              if (this.override) {
+                  this.centroidLayer.setVisibility(false);
+                  this.explodedLayer.forEach(function (layer) {
+                      layer.setVisibility(false);
+                  });
+                  return;
+              }
+
+              this.centroidLayer.setVisibility(extent.lod.level < this.scaleTrigger);
+              Object.keys(this.explodedLayer).forEach(function (key) {
+                  var layer = this.explodedLayer[key];
+                  layer.setVisibility(extent.lod.level >= this.scaleTrigger);
+              }, this);
+              // topic.publish(config.topics.layer.toggleVisibility, {
+              //     centroids: extent.lod.level < this.scaleTrigger,
+              //     projects: extent.lod.level >= this.scaleTrigger
+              // });
+          }));
+      },
+      _loadLayers: function () {
+          // summary:
+          //      adds the new layers if they haven't been yet
+          console.log('app.centroidSwitch::_loadLayers', arguments);
+
+          if (Object.keys(this.explodedLayer) === 0 || !this.centroidLayer) {
+              var li = config.layerIndices;
+              var deferreds = [];
+              var typesLookup = {
+                  0: 'poly-exploded',
+                  1: 'line-exploded',
+                  2: 'point-exploded'
+              };
+
+              [li.poly, li.line, li.point].forEach(function (layerIndex, i) {
+                  var layer = new FeatureLayer(config.urls.mapService + '/' + layerIndex, {
+                      id: typesLookup[i]
+                  });
+
+                  var deferred = new Deferred();
+
+                  layer.on('load', function () {
+                      deferred.resolve();
+                  });
+
+                  deferreds.push(deferred);
+
+                  this.explodedLayer[typesLookup[i]] = layer;
+              }, this);
+
+              this.centroidLayer = new FeatureLayer(config.urls.centroidService, {
+                  id: 'Centroid'
+              });
+
+              topic.publish(config.topics.layer.add, {
+                  graphicsLayers: [this.explodedLayer['poly-exploded'],
+                                   this.explodedLayer['line-exploded'],
+                                   this.explodedLayer['point-exploded'],
+                                   this.centroidLayer
+                                 ]
+              });
+
+              return all(deferreds);
+          }
+
+          var d = new Deferred();
```

Comment:
This is unneeded. I believe that you can just return `true` or something and then change `then` on 90 to `when`.